src/main/java/peoplesoft/model/employment/Employment.java

```diff
@@ -190,9 +225,9 @@ public Employment deserialize(JsonParser p, DeserializationContext ctx)
             }
 
             // readValueAs Map is ok because we know `node` has to be a json object
-            HashMap<ID, Name> map = node
+            Map<ID, Set<ID>> map = node
                 .traverse(codec)
-                .readValueAs(new TypeReference<HashMap<ID, Name>>(){});
+                .readValueAs(new TypeReference<Map<ID, Set<ID>>>(){});
 
             return new Employment(map);
```

Comment:
should we make a copy of the map & all the sets in the `Employment` constructor then? Jackson deserializes all `Map`s to `HashMap`, and all `Set`s` to `HashSet`s by default, so we're gonna get a mix of `HashSet`s and `TreeSet`s in there, i think

and should the constructor for `Employment` be private? are we gonna be creating new instances outside of this class (or nested classes), where we pass in maps to be stored in fields instead of creating a copy?

Reply:
Hmm good idea.

`Employment` is only public for tests, I can change it to package-private or something.

Reply:
Actually I need it to test `JobAssignCommand` too, so I don't think it can be not public sadly.